proto/tendermint/types/types.proto

```diff
@@ -102,6 +102,16 @@ message Vote {
   bytes validator_address = 6;
   int32 validator_index   = 7;
   bytes signature         = 8;
+  VoteExtension vote_extension = 9;
+}
+// VoteExtension is app-defined additional information to the validator votes.
+message VoteExtension {
+  bytes app_data_signed              = 1;
+  bytes app_data_self_authenticating = 2;
+}
+// VoteExtensionSigned is the part of VoteExtension that is actually signed by the validators.
+message VoteExtensionSigned {
```

Comment:
Both `VoteExtensionSigned` and `VoteExtension` have `app_data_signed`? I would add some more documentation to these two types so their domain and purpose is clear.